crates/shielded_token/src/vp.rs

```diff
@@ -624,20 +624,31 @@ where
                 return Err(error);
             }
         }
+        let masp_tx_id = shielded_tx.txid().into();
+        let has_trans_inputs = shielded_tx
+            .transparent_bundle()
+            .map(|b| !b.vin.empty())
+            .unwrap_or(false);
```

Comment:
nit: `.map(|...| <bool>).unwrap_or(false)` is a pattern that can be replaced by the more readable fn `.is_some_and(|...| <bool>)`